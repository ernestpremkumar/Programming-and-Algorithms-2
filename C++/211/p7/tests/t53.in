print
insert 3
print
insert 4
print
echo 7
breadth
print
echo 4
echo 7
echo 5
find 7
size
size
print
size
insert 2
size
echo 5
balanced
find 4
distance
echo 0
print
size
distance
print
find 1
find 6
balanced
print
size
breadth
find 5
insert 5
insert 3
find 4
print
insert 3
find 0
breadth
distance
size
echo 1
balanced
distance
insert 4
size
breadth
find 5
echo 6
size
find 3
distance
balanced
echo 2
find 4
distance
balanced
print
breadth
find 2
size
print
size
distance
find 3
distance
balanced
find 5
balanced
find 3
insert 2
echo 1
balanced
balanced
distance
find 2
balanced
insert 7
balanced
breadth
echo 2
insert 3
echo 0
distance
balanced
size
size
echo 2
balanced
distance
size
distance
echo 2
print
balanced
size
print
balanced
balanced
print
balanced
distance
insert 2
find 6
distance
breadth
size
find 4
find 3
size
echo 2
size
balanced
insert 1
balanced
print
print
insert 2
insert 5
breadth
distance
echo 3
balanced
find 3
size
breadth
size
balanced
echo 2
echo 0
distance
balanced
balanced
echo 0
print
echo 6
insert 2
find 0
print
size
print
insert 7
distance
insert 0
echo 7
balanced
distance
size
distance
distance
echo 2
breadth
balanced
insert 0
distance
insert 4
find 3
size
size
insert 3
print
size
distance
insert 4
distance
distance
distance
echo 3
find 5
echo 2
distance
breadth
print
find 7
size
insert 4
print
insert 3
balanced
distance
size
breadth
echo 0
breadth
balanced
distance
distance
balanced
size
insert 2
balanced
size
distance
distance
distance
find 4
insert 0
echo 0
distance
distance
size
find 4
balanced
find 2
print
breadth
balanced
balanced
find 5
size
insert 2
size
distance
breadth
echo 4
size
balanced
print
balanced
balanced
print
balanced
find 3
echo 6
find 7
insert 2
echo 4
balanced
distance
find 4
balanced
size
size
print
breadth
breadth
balanced
print
size
size
balanced
insert 3
distance
distance
breadth
print
balanced
find 6
print
find 0
find 1
find 0
print
distance
print
find 0
find 3
size
distance
distance
find 3
echo 2
echo 2
print
print
distance
echo 7
size
echo 5
find 0
echo 4
balanced
print
breadth
find 2
breadth
distance
echo 6
size
find 5
breadth
breadth
size
size
echo 2
balanced
breadth
print
echo 1
find 1
breadth
insert 0
distance
insert 2
balanced
distance
breadth
insert 4
print
size
size
breadth
balanced
echo 2
find 0
find 3
find 0
find 5
balanced
size
insert 6
size
breadth
distance
balanced
breadth
insert 1
echo 3
breadth
insert 0
find 0
insert 4
insert 7
insert 4
insert 0
distance
print
size
print
size
insert 3
breadth
find 2
find 5
distance
echo 4
insert 1
echo 1
distance
find 1
breadth
size
size
find 4
breadth
insert 5
breadth
echo 6
print
echo 4
echo 2
find 1
size
print
breadth
echo 7
echo 2
print
balanced
echo 7
print
size
echo 2
size
distance
distance
print
echo 2
distance
balanced
find 3
distance
insert 4
distance
insert 0
print
insert 5
balanced
insert 1
distance
breadth
distance
distance
balanced
find 4
echo 6
breadth
find 6
echo 7
balanced
echo 1
distance
size
find 6
echo 3
find 1
breadth
distance
echo 4
echo 5
print
find 0
distance
echo 6
balanced
balanced
print
insert 6
print
size
breadth
insert 5
distance
balanced
print
find 7
echo 0
size
print
breadth
size
find 4
size
breadth
insert 6
size
print
size
size
insert 1
echo 7
balanced
size
balanced
print
size
echo 6
breadth
size
insert 2
insert 4
find 4
balanced
balanced
distance
insert 1
breadth
balanced
size
distance
size
print
distance
breadth
find 0
print
find 4
insert 7
print
print
size
insert 5
distance
balanced
distance
distance
distance
size
print
breadth
balanced
print
insert 2
print
distance
echo 1
distance
echo 1
insert 4
print
breadth
balanced
breadth
echo 1
insert 1
echo 5
balanced
size
size
breadth
echo 0
insert 5
distance
find 1
echo 7
distance
size
size
balanced
echo 2
echo 0
insert 0
breadth
balanced
size
balanced
size
insert 0
insert 0
echo 4
find 2
echo 6
print
print
print
find 4
breadth
find 4
echo 2
insert 2
balanced
breadth
insert 1
insert 6
size
print
echo 5
find 4
distance
balanced
print
balanced
balanced
find 3
breadth
insert 4
print
breadth
print
echo 7
echo 6
print
insert 0
print
distance
size
distance
find 2
breadth
print
size
balanced
print
print
find 0
balanced
insert 4
print
size
insert 1
print
find 4
distance
balanced
breadth
breadth
insert 7
distance
insert 5
breadth
balanced
echo 4
balanced
size
distance
size
echo 2
breadth
find 6
insert 4
echo 0
echo 6
find 1
distance
insert 4
echo 3
echo 1
insert 1
distance
insert 4
echo 4
distance
echo 0
distance
echo 3
print
breadth
insert 4
size
print
size
breadth
distance
size
size
echo 3
distance
size
size
insert 0
balanced
size
insert 5
breadth
size
echo 2
balanced
balanced
find 5
insert 1
print
echo 1
balanced
echo 3
find 1
find 1
balanced
echo 5
echo 0
print
balanced
balanced
insert 6
find 6
balanced
find 4
breadth
breadth
distance
distance
distance
breadth
balanced
distance
size
size
balanced
echo 1
balanced
breadth
echo 6
insert 0
print
find 6
find 2
print
find 2
balanced
distance
find 5
echo 1
find 5
balanced
balanced
balanced
breadth
breadth
distance
breadth
print
size
size
size
insert 2
balanced
insert 4
insert 7
distance
echo 0
balanced
find 7
balanced
size
size
distance
echo 6
distance
balanced
print
insert 0
print
find 2
balanced
distance
echo 7
distance
insert 3
find 6
find 3
find 5
find 2
find 3
breadth
size
size
distance
distance
find 1
find 5
print
distance
distance
echo 7
find 2
balanced
distance
insert 3
breadth
find 6
echo 7
find 5
find 6
echo 1
distance
echo 3
balanced
print
distance
balanced
insert 1
balanced
size
echo 0
distance
distance
echo 7
breadth
size
insert 5
insert 0
size
echo 0
breadth
print
balanced
print
print
distance
breadth
balanced
find 3
breadth
size
distance
echo 3
breadth
balanced
insert 5
find 5
size
size
echo 6
print
size
size
print
echo 6
balanced
insert 2
distance
breadth
find 3
echo 4
insert 5
size
breadth
insert 5
find 5
find 5
distance
balanced
find 3
print
size
echo 5
distance
find 5
insert 2
balanced
distance
distance
distance
echo 0
breadth
echo 3
insert 4
balanced
balanced
balanced
insert 2
size
print
insert 6
insert 5
echo 0
size
find 7
balanced
size
balanced
echo 2
size
breadth
find 1
find 4
find 4
find 6
print
print
breadth
breadth
print
echo 1
breadth
balanced
breadth
breadth
balanced
echo 2
distance
distance
size
echo 2
balanced
find 4
insert 4
print
size
echo 4
distance
find 0
echo 3
size
size
size
balanced
insert 0
size
balanced
distance
distance
find 7
breadth
distance
print
breadth
echo 0
echo 5
breadth
breadth
find 7
balanced
breadth
distance
find 4
balanced
print
find 2
echo 1
distance
distance
print
insert 6
echo 5
echo 6
breadth
size
echo 5
balanced
print
find 2
print
distance
echo 7
insert 6
size
find 7
breadth
print
find 5
find 4
balanced
breadth
distance
balanced
balanced
distance
size
size
size
insert 3
size
find 4
insert 7
breadth
find 4
print
distance
find 2
distance
balanced
print
echo 5
insert 6
insert 3
distance
balanced
balanced
echo 3
size
print
distance
insert 0
size
insert 4
distance
distance
breadth
insert 7
size
insert 2
print
insert 0
find 3
find 2
balanced
print
echo 5
size
insert 2
find 3
find 7
insert 2
insert 0
distance
print
find 7
balanced
print
balanced
insert 0
balanced
print
size
size
insert 3
size
balanced
print
distance
find 2
insert 2
find 7
print
find 0
echo 3
insert 4
insert 7
balanced
find 3
distance
distance
echo 0
print
distance
distance
print
insert 7
echo 0
echo 5
echo 3
echo 0
balanced
balanced
breadth
balanced
balanced
print
print
print
size
print
size
distance
size
breadth
