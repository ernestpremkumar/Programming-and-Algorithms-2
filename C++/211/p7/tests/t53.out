{}
{3}
{3, 4}
7
{3, 4}
{3, 4}
4
7
5
<7> is not in tree.
2
2
{3, 4}
2
3
5
Tree is balanced.
<4> is in tree.
Average distance of nodes to root = 0.666667
0
{2, 3, 4}
3
Average distance of nodes to root = 0.666667
{2, 3, 4}
<1> is not in tree.
<6> is not in tree.
Tree is balanced.
{2, 3, 4}
3
{3, 2, 4}
<5> is not in tree.
<4> is in tree.
{2, 3, 4, 5}
<0> is not in tree.
{3, 2, 4, 5}
Average distance of nodes to root = 1
4
1
Tree is balanced.
Average distance of nodes to root = 1
4
{3, 2, 4, 5}
<5> is in tree.
6
4
<3> is in tree.
Average distance of nodes to root = 1
Tree is balanced.
2
<4> is in tree.
Average distance of nodes to root = 1
Tree is balanced.
{2, 3, 4, 5}
{3, 2, 4, 5}
<2> is in tree.
4
{2, 3, 4, 5}
4
Average distance of nodes to root = 1
<3> is in tree.
Average distance of nodes to root = 1
Tree is balanced.
<5> is in tree.
Tree is balanced.
<3> is in tree.
1
Tree is balanced.
Tree is balanced.
Average distance of nodes to root = 1
<2> is in tree.
Tree is balanced.
Tree is not balanced.
{3, 2, 4, 5, 7}
2
0
Average distance of nodes to root = 1.4
Tree is not balanced.
5
5
2
Tree is not balanced.
Average distance of nodes to root = 1.4
5
Average distance of nodes to root = 1.4
2
{2, 3, 4, 5, 7}
Tree is not balanced.
5
{2, 3, 4, 5, 7}
Tree is not balanced.
Tree is not balanced.
{2, 3, 4, 5, 7}
Tree is not balanced.
Average distance of nodes to root = 1.4
<6> is not in tree.
Average distance of nodes to root = 1.4
{3, 2, 4, 5, 7}
5
<4> is in tree.
<3> is in tree.
5
2
5
Tree is not balanced.
Tree is not balanced.
{1, 2, 3, 4, 5, 7}
{1, 2, 3, 4, 5, 7}
{3, 2, 4, 1, 5, 7}
Average distance of nodes to root = 1.5
3
Tree is not balanced.
<3> is in tree.
6
{3, 2, 4, 1, 5, 7}
6
Tree is not balanced.
2
0
Average distance of nodes to root = 1.5
Tree is not balanced.
Tree is not balanced.
0
{1, 2, 3, 4, 5, 7}
6
<0> is not in tree.
{1, 2, 3, 4, 5, 7}
6
{1, 2, 3, 4, 5, 7}
Average distance of nodes to root = 1.5
7
Tree is not balanced.
Average distance of nodes to root = 1.71429
7
Average distance of nodes to root = 1.71429
Average distance of nodes to root = 1.71429
2
{3, 2, 4, 1, 5, 0, 7}
Tree is not balanced.
Average distance of nodes to root = 1.71429
<3> is in tree.
7
7
{0, 1, 2, 3, 4, 5, 7}
7
Average distance of nodes to root = 1.71429
Average distance of nodes to root = 1.71429
Average distance of nodes to root = 1.71429
Average distance of nodes to root = 1.71429
3
<5> is in tree.
2
Average distance of nodes to root = 1.71429
{3, 2, 4, 1, 5, 0, 7}
{0, 1, 2, 3, 4, 5, 7}
<7> is in tree.
7
{0, 1, 2, 3, 4, 5, 7}
Tree is not balanced.
Average distance of nodes to root = 1.71429
7
{3, 2, 4, 1, 5, 0, 7}
0
{3, 2, 4, 1, 5, 0, 7}
Tree is not balanced.
Average distance of nodes to root = 1.71429
Average distance of nodes to root = 1.71429
Tree is not balanced.
7
Tree is not balanced.
7
Average distance of nodes to root = 1.71429
Average distance of nodes to root = 1.71429
Average distance of nodes to root = 1.71429
<4> is in tree.
0
Average distance of nodes to root = 1.71429
Average distance of nodes to root = 1.71429
7
<4> is in tree.
Tree is not balanced.
<2> is in tree.
{0, 1, 2, 3, 4, 5, 7}
{3, 2, 4, 1, 5, 0, 7}
Tree is not balanced.
Tree is not balanced.
<5> is in tree.
7
7
Average distance of nodes to root = 1.71429
{3, 2, 4, 1, 5, 0, 7}
4
7
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 7}
Tree is not balanced.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 7}
Tree is not balanced.
<3> is in tree.
6
<7> is in tree.
4
Tree is not balanced.
Average distance of nodes to root = 1.71429
<4> is in tree.
Tree is not balanced.
7
7
{0, 1, 2, 3, 4, 5, 7}
{3, 2, 4, 1, 5, 0, 7}
{3, 2, 4, 1, 5, 0, 7}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 7}
7
7
Tree is not balanced.
Average distance of nodes to root = 1.71429
Average distance of nodes to root = 1.71429
{3, 2, 4, 1, 5, 0, 7}
{0, 1, 2, 3, 4, 5, 7}
Tree is not balanced.
<6> is not in tree.
{0, 1, 2, 3, 4, 5, 7}
<0> is in tree.
<1> is in tree.
<0> is in tree.
{0, 1, 2, 3, 4, 5, 7}
Average distance of nodes to root = 1.71429
{0, 1, 2, 3, 4, 5, 7}
<0> is in tree.
<3> is in tree.
7
Average distance of nodes to root = 1.71429
Average distance of nodes to root = 1.71429
<3> is in tree.
2
2
{0, 1, 2, 3, 4, 5, 7}
{0, 1, 2, 3, 4, 5, 7}
Average distance of nodes to root = 1.71429
7
7
5
<0> is in tree.
4
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 7}
{3, 2, 4, 1, 5, 0, 7}
<2> is in tree.
{3, 2, 4, 1, 5, 0, 7}
Average distance of nodes to root = 1.71429
6
7
<5> is in tree.
{3, 2, 4, 1, 5, 0, 7}
{3, 2, 4, 1, 5, 0, 7}
7
7
2
Tree is not balanced.
{3, 2, 4, 1, 5, 0, 7}
{0, 1, 2, 3, 4, 5, 7}
1
<1> is in tree.
{3, 2, 4, 1, 5, 0, 7}
Average distance of nodes to root = 1.71429
Tree is not balanced.
Average distance of nodes to root = 1.71429
{3, 2, 4, 1, 5, 0, 7}
{0, 1, 2, 3, 4, 5, 7}
7
7
{3, 2, 4, 1, 5, 0, 7}
Tree is not balanced.
2
<0> is in tree.
<3> is in tree.
<0> is in tree.
<5> is in tree.
Tree is not balanced.
7
8
{3, 2, 4, 1, 5, 0, 7, 6}
Average distance of nodes to root = 2
Tree is not balanced.
{3, 2, 4, 1, 5, 0, 7, 6}
3
{3, 2, 4, 1, 5, 0, 7, 6}
<0> is in tree.
Average distance of nodes to root = 2
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
{3, 2, 4, 1, 5, 0, 7, 6}
<2> is in tree.
<5> is in tree.
Average distance of nodes to root = 2
4
1
Average distance of nodes to root = 2
<1> is in tree.
{3, 2, 4, 1, 5, 0, 7, 6}
8
8
<4> is in tree.
{3, 2, 4, 1, 5, 0, 7, 6}
{3, 2, 4, 1, 5, 0, 7, 6}
6
{0, 1, 2, 3, 4, 5, 6, 7}
4
2
<1> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
{3, 2, 4, 1, 5, 0, 7, 6}
7
2
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
7
{0, 1, 2, 3, 4, 5, 6, 7}
8
2
8
Average distance of nodes to root = 2
Average distance of nodes to root = 2
{0, 1, 2, 3, 4, 5, 6, 7}
2
Average distance of nodes to root = 2
Tree is not balanced.
<3> is in tree.
Average distance of nodes to root = 2
Average distance of nodes to root = 2
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 2
{3, 2, 4, 1, 5, 0, 7, 6}
Average distance of nodes to root = 2
Average distance of nodes to root = 2
Tree is not balanced.
<4> is in tree.
6
{3, 2, 4, 1, 5, 0, 7, 6}
<6> is in tree.
7
Tree is not balanced.
1
Average distance of nodes to root = 2
8
<6> is in tree.
3
<1> is in tree.
{3, 2, 4, 1, 5, 0, 7, 6}
Average distance of nodes to root = 2
4
5
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
Average distance of nodes to root = 2
6
Tree is not balanced.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
{3, 2, 4, 1, 5, 0, 7, 6}
Average distance of nodes to root = 2
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
<7> is in tree.
0
8
{0, 1, 2, 3, 4, 5, 6, 7}
{3, 2, 4, 1, 5, 0, 7, 6}
8
<4> is in tree.
8
{3, 2, 4, 1, 5, 0, 7, 6}
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
7
Tree is not balanced.
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
8
6
{3, 2, 4, 1, 5, 0, 7, 6}
8
<4> is in tree.
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 2
{3, 2, 4, 1, 5, 0, 7, 6}
Tree is not balanced.
8
Average distance of nodes to root = 2
8
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2
{3, 2, 4, 1, 5, 0, 7, 6}
<0> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
Average distance of nodes to root = 2
Tree is not balanced.
Average distance of nodes to root = 2
Average distance of nodes to root = 2
Average distance of nodes to root = 2
8
{0, 1, 2, 3, 4, 5, 6, 7}
{3, 2, 4, 1, 5, 0, 7, 6}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2
1
Average distance of nodes to root = 2
1
{0, 1, 2, 3, 4, 5, 6, 7}
{3, 2, 4, 1, 5, 0, 7, 6}
Tree is not balanced.
{3, 2, 4, 1, 5, 0, 7, 6}
1
5
Tree is not balanced.
8
8
{3, 2, 4, 1, 5, 0, 7, 6}
0
Average distance of nodes to root = 2
<1> is in tree.
7
Average distance of nodes to root = 2
8
8
Tree is not balanced.
2
0
{3, 2, 4, 1, 5, 0, 7, 6}
Tree is not balanced.
8
Tree is not balanced.
8
4
<2> is in tree.
6
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<4> is in tree.
{3, 2, 4, 1, 5, 0, 7, 6}
<4> is in tree.
2
Tree is not balanced.
{3, 2, 4, 1, 5, 0, 7, 6}
8
{0, 1, 2, 3, 4, 5, 6, 7}
5
<4> is in tree.
Average distance of nodes to root = 2
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
<3> is in tree.
{3, 2, 4, 1, 5, 0, 7, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
{3, 2, 4, 1, 5, 0, 7, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
7
6
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2
8
Average distance of nodes to root = 2
<2> is in tree.
{3, 2, 4, 1, 5, 0, 7, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
<4> is in tree.
Average distance of nodes to root = 2
Tree is not balanced.
{3, 2, 4, 1, 5, 0, 7, 6}
{3, 2, 4, 1, 5, 0, 7, 6}
Average distance of nodes to root = 2
{3, 2, 4, 1, 5, 0, 7, 6}
Tree is not balanced.
4
Tree is not balanced.
8
Average distance of nodes to root = 2
8
2
{3, 2, 4, 1, 5, 0, 7, 6}
<6> is in tree.
0
6
<1> is in tree.
Average distance of nodes to root = 2
3
1
Average distance of nodes to root = 2
4
Average distance of nodes to root = 2
0
Average distance of nodes to root = 2
3
{0, 1, 2, 3, 4, 5, 6, 7}
{3, 2, 4, 1, 5, 0, 7, 6}
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
{3, 2, 4, 1, 5, 0, 7, 6}
Average distance of nodes to root = 2
8
8
3
Average distance of nodes to root = 2
8
8
Tree is not balanced.
8
{3, 2, 4, 1, 5, 0, 7, 6}
8
2
Tree is not balanced.
Tree is not balanced.
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
1
Tree is not balanced.
3
<1> is in tree.
<1> is in tree.
Tree is not balanced.
5
0
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
<6> is in tree.
Tree is not balanced.
<4> is in tree.
{3, 2, 4, 1, 5, 0, 7, 6}
{3, 2, 4, 1, 5, 0, 7, 6}
Average distance of nodes to root = 2
Average distance of nodes to root = 2
Average distance of nodes to root = 2
{3, 2, 4, 1, 5, 0, 7, 6}
Tree is not balanced.
Average distance of nodes to root = 2
8
8
Tree is not balanced.
1
Tree is not balanced.
{3, 2, 4, 1, 5, 0, 7, 6}
6
{0, 1, 2, 3, 4, 5, 6, 7}
<6> is in tree.
<2> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<2> is in tree.
Tree is not balanced.
Average distance of nodes to root = 2
<5> is in tree.
1
<5> is in tree.
Tree is not balanced.
Tree is not balanced.
Tree is not balanced.
{3, 2, 4, 1, 5, 0, 7, 6}
{3, 2, 4, 1, 5, 0, 7, 6}
Average distance of nodes to root = 2
{3, 2, 4, 1, 5, 0, 7, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
8
Tree is not balanced.
Average distance of nodes to root = 2
0
Tree is not balanced.
<7> is in tree.
Tree is not balanced.
8
8
Average distance of nodes to root = 2
6
Average distance of nodes to root = 2
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<2> is in tree.
Tree is not balanced.
Average distance of nodes to root = 2
7
Average distance of nodes to root = 2
<6> is in tree.
<3> is in tree.
<5> is in tree.
<2> is in tree.
<3> is in tree.
{3, 2, 4, 1, 5, 0, 7, 6}
8
8
Average distance of nodes to root = 2
Average distance of nodes to root = 2
<1> is in tree.
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2
Average distance of nodes to root = 2
7
<2> is in tree.
Tree is not balanced.
Average distance of nodes to root = 2
{3, 2, 4, 1, 5, 0, 7, 6}
<6> is in tree.
7
<5> is in tree.
<6> is in tree.
1
Average distance of nodes to root = 2
3
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2
Tree is not balanced.
Tree is not balanced.
8
0
Average distance of nodes to root = 2
Average distance of nodes to root = 2
7
{3, 2, 4, 1, 5, 0, 7, 6}
8
8
0
{3, 2, 4, 1, 5, 0, 7, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2
{3, 2, 4, 1, 5, 0, 7, 6}
Tree is not balanced.
<3> is in tree.
{3, 2, 4, 1, 5, 0, 7, 6}
8
Average distance of nodes to root = 2
3
{3, 2, 4, 1, 5, 0, 7, 6}
Tree is not balanced.
<5> is in tree.
8
8
6
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
{0, 1, 2, 3, 4, 5, 6, 7}
6
Tree is not balanced.
Average distance of nodes to root = 2
{3, 2, 4, 1, 5, 0, 7, 6}
<3> is in tree.
4
8
{3, 2, 4, 1, 5, 0, 7, 6}
<5> is in tree.
<5> is in tree.
Average distance of nodes to root = 2
Tree is not balanced.
<3> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
8
5
Average distance of nodes to root = 2
<5> is in tree.
Tree is not balanced.
Average distance of nodes to root = 2
Average distance of nodes to root = 2
Average distance of nodes to root = 2
0
{3, 2, 4, 1, 5, 0, 7, 6}
3
Tree is not balanced.
Tree is not balanced.
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
0
8
<7> is in tree.
Tree is not balanced.
8
Tree is not balanced.
2
8
{3, 2, 4, 1, 5, 0, 7, 6}
<1> is in tree.
<4> is in tree.
<4> is in tree.
<6> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{3, 2, 4, 1, 5, 0, 7, 6}
{3, 2, 4, 1, 5, 0, 7, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
1
{3, 2, 4, 1, 5, 0, 7, 6}
Tree is not balanced.
{3, 2, 4, 1, 5, 0, 7, 6}
{3, 2, 4, 1, 5, 0, 7, 6}
Tree is not balanced.
2
Average distance of nodes to root = 2
Average distance of nodes to root = 2
8
2
Tree is not balanced.
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
8
4
Average distance of nodes to root = 2
<0> is in tree.
3
8
8
8
Tree is not balanced.
8
Tree is not balanced.
Average distance of nodes to root = 2
Average distance of nodes to root = 2
<7> is in tree.
{3, 2, 4, 1, 5, 0, 7, 6}
Average distance of nodes to root = 2
{0, 1, 2, 3, 4, 5, 6, 7}
{3, 2, 4, 1, 5, 0, 7, 6}
0
5
{3, 2, 4, 1, 5, 0, 7, 6}
{3, 2, 4, 1, 5, 0, 7, 6}
<7> is in tree.
Tree is not balanced.
{3, 2, 4, 1, 5, 0, 7, 6}
Average distance of nodes to root = 2
<4> is in tree.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
<2> is in tree.
1
Average distance of nodes to root = 2
Average distance of nodes to root = 2
{0, 1, 2, 3, 4, 5, 6, 7}
5
6
{3, 2, 4, 1, 5, 0, 7, 6}
8
5
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
<2> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2
7
8
<7> is in tree.
{3, 2, 4, 1, 5, 0, 7, 6}
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
<4> is in tree.
Tree is not balanced.
{3, 2, 4, 1, 5, 0, 7, 6}
Average distance of nodes to root = 2
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 2
8
8
8
8
<4> is in tree.
{3, 2, 4, 1, 5, 0, 7, 6}
<4> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2
<2> is in tree.
Average distance of nodes to root = 2
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
5
Average distance of nodes to root = 2
Tree is not balanced.
Tree is not balanced.
3
8
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2
8
Average distance of nodes to root = 2
Average distance of nodes to root = 2
{3, 2, 4, 1, 5, 0, 7, 6}
8
{0, 1, 2, 3, 4, 5, 6, 7}
<3> is in tree.
<2> is in tree.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
5
8
<3> is in tree.
<7> is in tree.
Average distance of nodes to root = 2
{0, 1, 2, 3, 4, 5, 6, 7}
<7> is in tree.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2
<2> is in tree.
<7> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
3
Tree is not balanced.
<3> is in tree.
Average distance of nodes to root = 2
Average distance of nodes to root = 2
0
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 2
Average distance of nodes to root = 2
{0, 1, 2, 3, 4, 5, 6, 7}
0
5
3
0
Tree is not balanced.
Tree is not balanced.
{3, 2, 4, 1, 5, 0, 7, 6}
Tree is not balanced.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
Average distance of nodes to root = 2
8
{3, 2, 4, 1, 5, 0, 7, 6}
